app/src/main/java/com/paddlesandbugs/dahdidahdit/brasspound/AbstractPaddleInputActivity.java

```diff
@@ -108,6 +110,17 @@ public boolean dispatchKeyEvent(KeyEvent event) {
         return super.dispatchKeyEvent(event);
     }
 
+    @Override
+    public boolean dispatchTouchEvent(MotionEvent event) {
+        if (event.getSource() != InputDevice.SOURCE_TOUCHSCREEN) {
+            morseInput.handleMouse(event); // otherwise taps on screen interfere; we only support EXTERNAL devices
+
+            return true;
```

Comment:
This return needs to be removed. If it stays, there is no way of using the app in an emulator.

Reply:
It's not possible afaik because the source when enabling hardware keyboard in the emulator will never be the touchscreen. So in the emulator you are stuck when using an external mouse. I do actually believe this is a bug in the emulator. If this return statement is indeed removed my paddle closes the app (because it uses a secondary button click = right click = back on my pixel).

Reply:
I'll have a look if I can fix this in another way again

Reply:
Oh, that sounds fun :frowning_face: 

Maybe we could add a workaround by determining if the code is running in an emulator:
https://stackoverflow.com/questions/6864251/how-to-find-out-from-code-if-my-android-app-runs-on-emulator-or-real-device

Reply:
I think this is not solvable - mouse support can never work in the emulator. Here is the reason:

* When you click in the emulator with hardware forwarding disabled you get source === SOURCE_TOUCHSCREEN.
* When you enable hardware forwarding even simple clicks now have another source (TRACKPAD or MOUSE iirc)
* Your mouse cannot be distinguished in the emulator anymore

Do you have any other idea?